dockerfiles/app/Dockerfile

```diff
@@ -1,15 +1,17 @@
-ARG BASE_IMAGE
-ARG IMAGE_TAG=latest
+ARG BASE_IMAGE=node
+ARG IMAGE_TAG=14-alpine
 FROM ${BASE_IMAGE}:${IMAGE_TAG}
 
 WORKDIR /opt/app
 
-COPY . .
+# use changes to package.json to force Docker not to use the cache
+# when we change our application's nodejs dependencies:
+COPY package.json package.json
```

Comment:
Add https://stackoverflow.com/questions/35774714/how-to-cache-the-run-npm-install-instruction-when-docker-build-a-dockerfile